src/util/util.ts

```diff
@@ -108,7 +108,9 @@ const evalReg = new RegExp(/\beval\(([^),]*)\)/g);
 
 // hasEval determine whether matcher contains function eval
 function hasEval(s: string): boolean {
-  return evalReg.test(s);
+  const res: boolean = evalReg.test(s);
+  evalReg.lastIndex = 0;
```

Comment:
@Zxilly , @Shivansh-yadav13 
I'm not familiar with node-casbin but is /g flag really needed? Removing it would resolve the issue.

Reply:
you are right we don't need the `g` flag for this purpose but it is being utilized in another function(`getEvalValue`) to get all the matches.

Reply:
You can have two regex expressions

Reply:
maybe we can directly return like this then?
```js
return /\beval\(([^),]*)\)/.test(s)
```
@Zxilly @grzegorz-kap

Reply:
You would compile the regex expression during each invocation. It might have an impact on performance.

Reply:
> You can have two regex expressions

ok